The null hypothesis assumes that the actual operating costs are in line with the theoretical model or lower.

The alternative hypothesis claims that the actual operating costs are higher than the theoretical model.





In [1]:
# Hypotheses:
# Null Hypothesis (H₀): The weekly operating cost is not higher than the theoretical cost model (W = $1000 + $5X).
# Alternative Hypothesis (H₁): The weekly operating cost is higher than the theoretical cost model (W = $1000 + $5X).

import math

# Given values
sample_mean = 3050  # Sample mean (Rs. 3050)
theoretical_mean = 1000 + 5 * 600  # Theoretical mean (W = 1000 + 5 * X, where X = 600 units)
std_dev = 5 * 25  # Standard deviation (5 * 25)
n = 25  # Sample size

# Calculate the test statistic (t)
t_statistic = (sample_mean - theoretical_mean) / (std_dev / math.sqrt(n))

# Display the test statistic
print(f"Test Statistic (t): {t_statistic}")


Test Statistic (t): -38.0


* Sample Mean: This is the observed average operating cost across 25 restaurants, which is Rs. 3050.
* Theoretical Mean: This is the operating cost based on the theoretical model, calculated as
1000 + 5 × 600 = Rs 4000.
* Standard Deviation: Since each unit's cost follows a normal distribution with a standard deviation of 25 units, the total standard deviation for the operating cost is 5×25=Rs 125.
* Test Statistic (t): This measures how far the sample mean deviates from the theoretical mean, normalized by the standard deviation of the sample mean.

In [2]:
from scipy.stats import norm

# Set alpha level
alpha = 0.05

# Find critical value for one-tailed test
critical_value = norm.ppf(1 - alpha)

# Display the critical value
print(f"Critical Value (Z): {critical_value}")


Critical Value (Z): 1.6448536269514722


* Significance Level (α = 0.05): This is the threshold used to determine whether the test statistic is extreme enough to reject the null hypothesis.
* Critical Value (Z): The critical value is obtained from the Z-distribution for a one-tailed test with a 5% significance level. For this test, the critical value is approximately 1.645.

In [3]:
# Compare test statistic with critical value
if t_statistic > critical_value:
    print("Reject the null hypothesis: The weekly operating cost is higher than the theoretical model suggests.")
else:
    print("Fail to reject the null hypothesis: There is no significant evidence to suggest an increase in weekly operating costs.")


Fail to reject the null hypothesis: There is no significant evidence to suggest an increase in weekly operating costs.


Decision:
* If the test statistic is larger than the critical value, you reject the null hypothesis, which suggests that the actual weekly operating costs are indeed higher than the theoretical model.
* If the test statistic is smaller than or equal to the critical value, you fail to reject the null hypothesis, indicating that the observed data does not provide enough evidence to support the claim that costs have increased.

# Conclusion:
# Based on the calculated test statistic and the critical value, we conclude whether the restaurant owners' claim
# is supported by the data. If the null hypothesis is rejected, this indicates that the weekly operating costs are
# indeed higher than what the model predicts, supporting the restaurant owners' feedback.
